File: 20221220/WordChain.test.js
import solution, { calcNumberAndOrder } from './WordChain';

test('solution', () => {
  expect(solution(3, ['tank', 'kick', 'know', 'wheel',
    'land', 'dream', 'mother', 'robot', 'tank'])).toEqual([3, 3]);

  expect(solution(5, ['hello', 'observe', 'effect', 'take', 'either',
    'recognize', 'encourage', 'ensure', 'establish', 'hang', 'gather',
    'refer', 'reference', 'estimate', 'executive'])).toEqual([0, 0]);

  expect(solution(2, ['hello', 'one', 'even', 'never', 'now', 'world', 'draw']))
    .toEqual([1, 3]);
});

test('calcNumberAndOrder', () => {
  expect(calcNumberAndOrder(8, 3)).toEqual([3, 3]);
  expect(calcNumberAndOrder(4, 2)).toEqual([1, 3]);
});

File: 2022/20221219/wordChain/src/test/java/WordChainTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class WordChainTest {
    @Test
    void solution() {
        WordChain test = new WordChain();

        assertArrayEquals(new int[]{3, 3}, test.solution(
                3, new String[]{"tank", "kick", "know", "wheel", "land", "dream", "mother", "robot", "tank"}));

        assertArrayEquals(new int[]{0, 0}, test.solution(
                5, new String[]{"hello", "observe", "effect", "take", "either", "recognize", "encourage", "ensure",
                        "establish", "hang", "gather", "refer", "reference", "estimate", "executive"}));

        assertArrayEquals(new int[]{1, 3}, test.solution(
                2, new String[]{"hello", "one", "even", "never", "now", "world", "draw"}));
    }
}
